flink-staging/flink-hcatalog/src/main/java/org/apache/flink/hcatalog/java/HCatOutputFormat.java

```diff
@@ -0,0 +1,126 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.hcatalog.java;
+
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.java.tuple.Tuple;
+import org.apache.flink.api.java.typeutils.TupleTypeInfo;
+import org.apache.flink.hcatalog.HCatOutputFormatBase;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hive.hcatalog.data.DefaultHCatRecord;
+import org.apache.hive.hcatalog.data.HCatRecord;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/*
+ * A OutputFormat to write to HCatalog tables
+ * The outputFormat supports write to table and partitions through partitionValue map.
+ *
+ * Flink {@link org.apache.flink.api.java.tuple.Tuple} and {@link HCatRecord} are accepted
+ * as data source. If user want to write other type of data to HCatalog, a preceding Map function
+ * is required to convert the user data type to {@link HCatRecord}
+ *
+ * The constructor provides type checking if the user also pass a typeinfo object for the tuple type
+ */
+public class HCatOutputFormat<T> extends HCatOutputFormatBase<T> {
+
+
+	/**
+	 * Create HCatOutputFormat with default conf
+	 * @param database The database schema for the hcatalog table
+	 * @param table the name of the table
+	 * @param partitionValues Map of partition values, if null, whole table is used
+	 * @throws IOException
+	 */
+	public HCatOutputFormat(String database, String table, Map<String, String> partitionValues
+							) throws IOException
+	{
+		super(database, table, partitionValues);
+	}
+
+	/**
+	 * Create HCatOutputFormat with given conf
+	 * @param database The database schema for the hcatalog table
+	 * @param table the name of the table
+	 * @param partitionValues Map of partition values, if null, whole table is used
+	 * @param conf a hadoop configuration
+	 * @throws IOException
+	 */
+	public HCatOutputFormat(String database, String table, Map<String, String> partitionValues,
+							Configuration conf) throws IOException
+	{
+		super(database, table, partitionValues, conf);
+	}
+
+	/**
+	 * Create HCatOutputFormat with type checking
+	 * @param database The database schema for the hcatalog table
+	 * @param table the name of the table
+	 * @param partitionValues Map of partition values, if null, whole table is used
+	 * @param conf a hadoop configuration
+	 * @param typeInfo a typeInfo to be matched by the HCatSchema
+	 * @throws IOException
+	 */
+	public HCatOutputFormat(String database, String table, Map<String, String> partitionValues,
+							Configuration conf, TypeInformation<T> typeInfo) throws IOException
```

Comment:
Because of using tab character for indentation, inserting newline before starting first parameter would be better.